tests/fsharp/core/subtype/test.fsx

```diff
@@ -1696,6 +1696,25 @@ module RecordPropertyConstraintTests =
     check "ckjwnewk" (f8()) (System.TimeSpan.FromSeconds 2.0) // after mutation
     check "ckjwnewk" (f10()) "Gary"
 
+// See https://github.com/Microsoft/visualfsharp/issues/740 - inlining on subtypes was not allowed
```

Comment:
This test case looks the same as https://github.com/Microsoft/visualfsharp/pull/763/files#diff-e2959d5dc5cc275246c938d409549ee1R1699, which is added as a succeeding test case.  Have you checked that this test case still gives the errors shown after the fix is made?